### DISCUSSION TITLE
Trading between players

### DISCUSSION THREAD

**Discussion by Theod0reWu** [bcollazo/catanatron#255] [Q&A]
Hello,

This looks like an interesting project. There doesn't appear to any trade deals proposed between bots when they are playing. Is this functionality added? If not, are there any plans to add it?

I'd argue that trading can have a significant impact on the game and any bot that is going to incorporated online against players will need to be able to respond to trade deals.

**Comment by bcollazo** [comment]
Hey Theod0reWu, you are right! Right now the bots we have (Random, WeightedRandom, AlphaBeta, ...) don't make OFFER_TRADE plays. This is on purpose right now, because all the bots are structured to make decisions by analyze the game as a state tree, and the trading mechanism adds a lot of complexity to that state tree (for example, the tree becomes infinite and it can have loops). And since even in the "simpler version" of the game without trade offers we haven't achieved superhuman play in my opinion, then we are yet to introduce that to the mix.

That being said, the mechanism/game-logic is there thanks to: https://github.com/bcollazo/catanatron/pull/206. If you implement a bot that decides to OFFER_TRADE at any turn, the rest of the bots should at least respond to the actions to the best of their ability. That is, the bots right now can answer to trades, the infrastructure is there, but they don't have logic to generate OFFER_TRADEs reasonably.

I agree with you that trading is important and before we can claim superhuman play in complete catan, a strong bot should be able to smartly think about trades.
